bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/ValueDecoder.java

```diff
@@ -0,0 +1,384 @@
+            String value = new SimpleDateFormat(DateTimeType.DATE_PATTERN).format(cal.getTime());
+            return DateTimeType.valueOf(value);
+        } else if (translatorDateTime.isValidField(DPTXlatorDateTime.YEAR)
+                && translatorDateTime.isValidField(DPTXlatorDateTime.TIME)) {
+            // Date format and time information
+            cal.setTimeInMillis(translatorDateTime.getValueMilliseconds());
+            String value = new SimpleDateFormat(DateTimeType.DATE_PATTERN).format(cal.getTime());
+            return DateTimeType.valueOf(value);
+        } else {
+            LOGGER.warn("Failed to convert '{}'", translator.getValue());
+            return null;
+        }
+    }
+
+    private static @Nullable Type handleDpt232(String value, String subType) {
+        Matcher rgb = RGB_PATTERN.matcher(value);
+        if (rgb.matches()) {
+            int r = Integer.parseInt(rgb.group("r"));
+            int g = Integer.parseInt(rgb.group("g"));
+            int b = Integer.parseInt(rgb.group("b"));
+
+            switch (subType) {
+                case "600":
+                    return HSBType.fromRGB(r, g, b);
+                case "60000":
+                    // MDT specific: mis-use 232.600 for hsv instead of rgb
+                    DecimalType hue = new DecimalType(coerceToRange(r * 360.0 / 255.0, 0.0, 359.9999));
+                    PercentType sat = new PercentType(BigDecimal.valueOf(coerceToRange(g / 2.55, 0.0, 100.0)));
+                    PercentType bright = new PercentType(BigDecimal.valueOf(coerceToRange(b / 2.55, 0.0, 100.0)));
+                    return new HSBType(hue, sat, bright);
+                default:
+                    LOGGER.warn("Unknown subtype '232.{}', no conversion possible.", subType);
+                    return null;
+            }
+        }
+        LOGGER.warn("Failed to convert '{}' (DPT 232): Pattern does not match", value);
+        return null;
+    }
+
+    private static @Nullable Type handleDpt242(String value) {
+        Matcher xyY = XYY_PATTERN.matcher(value);
+        if (xyY.matches()) {
+            String stringx = xyY.group("x");
+            String stringy = xyY.group("y");
+            String stringY = xyY.group("Y");
+
+            if (stringx != null && stringy != null) {
+                double x = Double.parseDouble(stringx.replace(",", "."));
+                double y = Double.parseDouble(stringy.replace(",", "."));
+                if (stringY == null) {
+                    return HSBType.fromXY((float) x, (float) y);
+                    // return ColorUtil.xyToHsv(new double[] { x, y });
+                } else {
+                    // FIXME core does not support xyY yet
+                    return HSBType.fromXY((float) x, (float) y);
+                    // double Y = Double.parseDouble(stringY.replace(",", "."));
+                    // return ColorUtil.xyToHsv(new double[] { x, y, Y });
```

Comment:
Comment for remembering about this.

Reply:
waiting for openhab-core/3434 to be merged....
Does it make sense to catch java.lang.NoSuchMethodError in case someone deploys this to 4.0.0M1?

Reply:
Now don't provide compatibility for such a case and it would be nearly impossible to write workarounds for all of these cases.